### PR TITLE
feat(overthebox.autoconfigure): add dhcp autoconfigure for otb

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-431
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

## Description

For OverTheBox, add new tab for Configuration with the first elements available to configure: DHCP and Static lease

### CODE REVIEW

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
+export default class OverTheBoxAutoconfigureDhcp {
+  /* @ngInject */
+  constructor($http, $q, $state, $translate) {
+    this.$http = $http;
+    this.$q = $q;
+    this.$state = $state;
```
Comment (radireddy) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js:6]: $state is not used, remove it.

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/overTheBox-autoconfigure.controller.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export default class OverTheBoxAutoconfigureCtrl {
```
Comment (radireddy) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/overTheBox-autoconfigure.controller.js:1]: this controller not required. Delete it.

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/index.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import component from './autoconfigure-dhcp.component';
+
+const moduleName = 'ovhManagerOtbAutoconfigureDhcp';
+
+angular
+  .module(moduleName, ['ui.router'])
+  .component('autoconfigureDhcp', component);
```
Comment (antleblanc) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/index.js:7]: **issue(blocking):** Component name is too generic. For consistency, please keep `overTheBox` as component prefix.

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+                        <input
+                            class="oui-input"
+                            name="interface"
+                            type="text"
+                            ng-model="$ctrl.config.interface"
+                        />
+                    </oui-field>
+                    <oui-field
+                        class="ml-2"
+                        label="{{:: 'overTheBox_autoconfigure_dhcp_lease_duration' | translate}} *"
```
Comment (antleblanc) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html:44]: **nitpick:**  Please avoid using `*` to mark field as required. Prefer using `(optional)` suffix wherever is needed.

```suggestion
                        label="{{:: 'overTheBox_autoconfigure_dhcp_lease_duration' | translate}}"
```

**Review by antleblanc — DISMISSED** [review]

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+  createStaticLease() {
+    this.errorMessage = '';
+    this.successMessage = '';
+    if (
+      !this.staticLease.hostname &&
+      !this.staticLease.ip &&
+      !this.staticLease.mac &&
+      !this.staticLease.priority
+    ) {
+      return null;
```
Comment (marie-j) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js:185]:
```suggestion
      return this.$q.resolve(null);
```
to be consistent with return type if chain is needed

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+                property="priority"
+            >
+            </oui-datagrid-column>
+            <oui-datagrid-column>
+                <oui-button
+                    variant="primary"
+                    size="s"
+                    on-click="$ctrl.removeConfig($row)"
+                >
+                    <span class="oui-icon oui-icon-trash"></span>
```
Comment (marie-j) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html:158]:
```suggestion
                    <span class="oui-icon oui-icon-trash" aria-hidden="true"></span>
```
Also we should add a `sr-only` description

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/template.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<div class="overthebox">
+    <div class="telecom-section-content">
+        <div class="widget-presentation">
```
Comment (marie-j) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/template.html:3]: No alternative in ui-kit ?

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  "overTheBox_autoconfigure_delete": "Supprimer",
+  "overTheBox_autoconfigure_apply": "Appliquer",
+  "overTheBox_autoconfigure_add": "Ajouter",
+  "overTheBox_autoconfigure_submit": "Valider",
+  "overTheBox_autoconfigure_cancel": "Annuler",
+  "overTheBox_autoconfigure_no_values": "Il n'y a pas de valeurs définies pour cette section.",
+  "overTheBox_autoconfigure_loading_failed": "Une erreur s'est produite lors de la récupération des données : {{errorMessage}}.",
+  "overTheBox_autoconfigure_create_failed": "Une erreur s'est produite lors de la création : {{errorMessage}}.",
+  "overTheBox_autoconfigure_remove_failed": "Une erreur s'est produite lors de la suppression : {{errorMessage}}.",
+  "overTheBox_autoconfigure_mandatory": "Ce champs est obligatoire"
```
Comment (marie-j) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/translations/Messages_fr_FR.json:32]:
```suggestion
  "overTheBox_autoconfigure_mandatory": "Ce champ est obligatoire"
```

**Review by marie-j — DISMISSED** [review]

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/template.html (hunk 7 above)
Comment (Steffy29) [reply packages/manager/modules/overthebox/src/overthebox/autoconfigure/template.html:3]: No sorry.

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js (hunk 5 above)
Comment (Steffy29) [reply packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js:185]: There is no chain needed.

Comment (marie-j) [reply packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js:185]: Even though, it is nice to be consistent in return type

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+  }
+
+  loadStaticLeases() {
+    this.staticLeases = [];
+    return this.$http
+      .get(`/overTheBox/${this.serviceName}/configuration/dhcp/staticLease`)
+      .then((response) => {
+        if (response.data) {
+          this.staticLeases = response.data;
+        }
```
Comment (JeremyDec) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js:153]: `response.data` should be at least empty, and if no `response.data`, we are in error case no?
```suggestion
          this.staticLeases = response.data;
```

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js [hunk 10]
```diff
@@ -0,0 +1,256 @@
...
+  }
+
+  loadConfig() {
+    this.configList = [];
+    return this.$http
+      .get(`/overTheBox/${this.serviceName}/configuration/dhcp/config`)
+      .then((response) => {
+        if (response.data) {
+          this.configList = response.data;
+        }
```
Comment (JeremyDec) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js:40]: Same here:
```suggestion
        this.configList = response.data;
```

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/index.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import component from './autoconfigure-dhcp.component';
+
+const moduleName = 'ovhManagerOtbAutoconfigureDhcp';
+
+angular
+  .module(moduleName, ['ui.router'])
+  .component('otbAutoconfigureDhcp', component);
```
Comment (antleblanc) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/index.js:7]:
```suggestion
  .component('overTheBoxAutoconfigureDhcp', component);
```
For consistency:
```sh
$ git grep "\.component('" -- "packages/manager/modules/overthebox/"
packages/manager/modules/overthebox/src/overthebox/actions/index.js:  .component('overTheBoxActions', component)
packages/manager/modules/overthebox/src/overthebox/configure/index.js:  .component('overTheBoxConfigure', component)
packages/manager/modules/overthebox/src/overthebox/details/index.js:  .component('overTheBoxDetails', component)
packages/manager/modules/overthebox/src/overthebox/logs/index.js:  .component('overTheBoxLogs', component)
packages/manager/modules/overthebox/src/overthebox/order/index.js:  .component('overTheBoxOrder', component)
packages/manager/modules/overthebox/src/overthebox/overTheBox.module.js:  .component('ovhManagerOverTheBoxComponent', component)
packages/manager/modules/overthebox/src/overthebox/remote/index.js:  .component('overTheBoxRemote', component)
packages/manager/modules/overthebox/src/overthebox/tasks/index.js:  .component('overTheBoxTasks', component)
packages/manager/modules/overthebox/src/overtheboxes.module.js:  .component('ovhManagerOverTheBoxes', component)
```

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html [hunk 12]
```diff
@@ -0,0 +1,290 @@
...
+    <div class="oui-box oui-box_light">
+        <h4
+            class="oui-box__heading"
+            data-translate="overTheBox_autoconfigure_dhcp_config"
+        ></h4>
+        <div data-ng-if="$ctrl.configList.length === 0" class="mb-2">
+            <em data-translate="overTheBox_autoconfigure_no_values"></em>
+        </div>
+        <div data-ng-if="$ctrl.showAddConfig">
+            <form novalidate name="formAddConf">
```
Comment (antleblanc) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html:29]: **nitpick:** Parent `div` is not required. ngIf can be applied directly on the `<form>` element.

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html [hunk 13]
```diff
@@ -0,0 +1,290 @@
...
+        <oui-button
+            variant="primary"
+            data-on-click="$ctrl.addConfig()"
+            class="mb-2 d-flex justify-content-center"
+            data-ng-if="$ctrl.displayAddConfigButton"
+        >
+            <span
+                data-translate="overTheBox_autoconfigure_dhcp_config_add"
+            ></span>
+        </oui-button>
```
Comment (antleblanc) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html:122]: **question:** Why not prefer using `<oui-datagrid-topbar>`?
see: https://ovh.github.io/ovh-ui-kit/?path=/story/design-system-components-datagrid-webcomponent--with-topbar

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html [hunk 14]
```diff
@@ -0,0 +1,290 @@
...
+    <div class="oui-box oui-box_light mt-2">
+        <h4
+            class="oui-box__heading"
+            data-translate="overTheBox_autoconfigure_dhcp_static_leases"
+        ></h4>
+        <div data-ng-if="$ctrl.staticLeases.length === 0" class="mb-2">
+            <em data-translate="overTheBox_autoconfigure_no_values"></em>
+        </div>
+        <div data-ng-if="$ctrl.showAddStaticLease">
+            <form novalidate name="formStaticLease">
```
Comment (antleblanc) [inline packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html:175]: **nitpick:** Parent `div` is not required. ngIf can be applied directly on the `<form>` element.

**Review by antleblanc — CHANGES_REQUESTED** [review]

File: packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html (hunk 14 above)
Comment (antleblanc) [reply packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/template.html:187]: Can we move the `ngIf` directly on the `<form>` element? (and get rid of the `<div>` element).
Thanks

**Review by antleblanc — DISMISSED** [review]

**Review by antleblanc — APPROVED** [review]

**Review by antleblanc — DISMISSED** [review]

**Review by JeremyDec — APPROVED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**